### PR TITLE
fix(core): fix validateBindMfaBackupCode to force linking backup code

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Fix `validateBindMfaBackupCode` to align with the new logic: if backup code is enabled in Console, every user who has other MFA should setup backup code before signing in.

## Testing

Unit tests covered several cases.

## Checklist

~~- [ ] `.changeset`~~
~~- [x] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts [hunk 1]
```diff
@@ -222,16 +222,31 @@ export const validateBindMfaBackupCode = async (
 
   if (
     !factors.includes(MfaFactor.BackupCode) ||
-    bindMfas.length === 0 ||
     bindMfas.some(({ type }) => type === MfaFactor.BackupCode)
   ) {
     return interaction;
   }
 
+  // Skip check if there is no new normal MFA
```
Comment (darcyYe) [inline packages/core/src/routes/interaction/verifications/mfa-verification.ts:230]: normal MFA means TOTP and Webauthn?

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts [hunk 2]
```diff
@@ -222,16 +222,31 @@ export const validateBindMfaBackupCode = async (
...
+  if (
+    event === InteractionEvent.Register &&
+    !bindMfas.some(({ type }) => type !== MfaFactor.BackupCode)
+  ) {
+    return interaction;
+  }
```
Comment (simeng-li) [inline packages/core/src/routes/interaction/verifications/mfa-verification.ts:236]: move this blow `if (event === InteractionEvent.SignI)` and only check 

```
if (!bindMfas.find(({ type }) => type !== MfaFactor.BackupCode)) {
  return interaction;
}
```

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/verifications/mfa-verification.ts]: Yes, updated to "other MFA"

File: packages/core/src/routes/interaction/verifications/mfa-verification.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/interaction/verifications/mfa-verification.ts]: updated

**Review by simeng-li — APPROVED** [review]

**Pull request merged**